### ISSUE TITLE
AssertError in tests.config.TestControlledDict when run on Python 3.11

### ISSUE THREAD

**Issue by cheese** [zim-desktop-wiki/zim-desktop-wiki#2087]
**Describe the bug**

AssertError in tests.config.TestControlledDict when run on Fedora Rawhide(37) Python 3.11

```
FAIL: runTest (tests.config.TestControlledDict.runTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/zim-0.74.3/tests/config.py", line 256, in runTest
    self.assertTrue(mydict.modified)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: False is not true
```
Full build log: https://kojipkgs.fedoraproject.org//work/tasks/2658/88342658/build.log

**To Reproduce**
1. run `LANG=en_US.UTF-8 xvfb-run ./test.py` under Python 3.11

**Expected behavior**
Tests all pass.

**Environment (please complete the following information):**
 - Version [e.g. 0.74.0]: 0.74.3
 - OS: [e.g. linux/windows/mac os] Fedora 37 with Python 3.11
 - Language: en_US.UTF-8

**Comment by introt** [comment]
The test corresponds to the `pop` method of [ControlledDict](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/zim/config/dicts.py#L46). `git diff` shows no relevant changes in `develop`, so I'd assume it is also affected.

**Comment by oturpe** [comment]
This is related to [Python Issue27275](https://bugs.python.org/issue27275).
I am not a Python expert, but apparently,
`OrderedDict.pop()` was misbehaving by calling `__delitem__`,
and that has now been fixed for Python 3.11.

On the other hand, ControlledDict and its test suite expect `__delitem__` call to happen.
I suppose this could be fixed by overriding _pop()_ in _ControlledDict_,
so that it emits the _changed_ event separately.
